### PR TITLE
DOC-3299 - Update OLX Guide: add levels of support for problem types, reorganize

### PR DESCRIPTION
**Pull request by catong**
## [DOC-3299](https://openedx.atlassian.net/browse/DOC-3299)

This PR reorganizes the existing "Components (XBlocks)" and "Problems" sections of the OLX Guide.
- The "problem-components" subsection was moved to "Components" to follow the parallel HTML, discussion, and video component subsections.
- All individual topics for xblocks and problems, previously divided under "Components (XBlocks)" and "Problems", were consolidated under the "Problems" section.
- The "Problems" section was renamed to "Exercises, Tools, and Problem Types". 
- A new OLX-only file "Levels of Support" was added to the "Exercises, Tools, and Problem Types" section.
### Reviewers

The PR submitter checks the boxes after each reviewer finishes and gives :+1:. 
- [x] Doc team review (sanity check): @lamagnifica or @srpearce 

FYI: @marcotuts @sstack22 @cahrens 
### Testing
- [x] Ran ./run_tests.sh without warnings or errors
### HTML Version (optional)
- [x] http://draft-supportlevels.readthedocs.io/en/latest/
### Post-review
- [ ] Add a comment with the description of this change or link this PR to the next release notes task.
- [x] Squash commits

### CODE REVIEW

File: en_us/olx/source/components/problem-components.rst [hunk 1]
```diff
@@ -1,281 +1,311 @@
...
+   must make are minor, duplicate the problem component before you delete it
+   and revise the copy.) Then ask all your learners to complete the new
+   problem.
+
+
+.. _Multiple Problems in One Component:
+
+***********************************
+Multiple Problems in One Component
+***********************************
```
Comment (lamagnifica) [inline en_us/olx/source/components/problem-components.rst:235]: update this topic to match what is now in the Studio version of this file? Or, maybe just refer to that section without repeating all of it? none of this is OLX specific...

Comment (catong) [reply en_us/olx/source/components/problem-components.rst:235]: @lamagnifica Totally agree that more work needs to be done to correctly share the content between OLX and BR guides. However, can we keep this PR fairly contained for sanity? Please check this PR for sanity and coherence, and once this merges I will create follow up stories to update the content that should be shared. Thanks!

File: en_us/olx/source/components/problem-components.rst [hunk 2]
```diff
@@ -1,281 +1,311 @@
-.. _Working with Problems:
-
-################################
-Working with Problem Components
-################################
-
-This section covers the basics of problem components: what they look like to
-you and your learners, and the options that every problem component has.
```
Comment (lamagnifica) [inline en_us/olx/source/components/problem-components.rst:8]: oddly, I cannot locate this info in the currently published version of the OLX guide

File: en_us/olx/source/components/problem-components.rst (hunk 1 above)
Comment (lamagnifica) [reply en_us/olx/source/components/problem-components.rst:235]: cool

File: en_us/olx/source/problem-xml/index.rst [hunk 3]
```diff
@@ -1,65 +1,55 @@
...
-The most general grader is ``customresponse``. This grader uses Python code to
-evaluate the input. By design, this ought to work with any inputtype, although
-there are bugs mixing this with a small number of the newer inputtypes.
-
-Like LON-CAPA, OLX allows embedding of code to generate parameterized problems.
-Unlike LON-CAPA, edX supports Python (and not Perl). Otherwise, the syntax for
-parameterizing problems is approximately identical.
 
 .. toctree::
    :maxdepth: 2
```
Comment (lamagnifica) [inline en_us/olx/source/problem-xml/index.rst]: maybe change to maxdepth: 1 -- the toc is reallllly long

File: en_us/olx/source/problem-xml/levels_of_support.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+.. list-table::
+   :widths: 25 60
+   :header-rows: 1
+
+   * - Level of Support
+     - Description
+   * - Full support
+     - Fully supported tools and features are available on edx.org, are fully
+       tested, have user interfaces where applicable, and are documented in the
+       official edX guides that are available on docs.edx.org.
```
Comment (lamagnifica) [inline en_us/olx/source/problem-xml/levels_of_support.rst]: link?

**Comment by lamagnifica** [comment]
👍 all files present and accounted for after reorg
There is some weirdness about what I am seeing in the published guide vs. in the deleted lines in this PR. Possibly a rebase is needed.
Please open a follow on story to strip all possible detail out of this guide and instead redirect readers to the B&R guides for info.

File: en_us/olx/source/components/problem-components.rst (hunk 2 above)
Comment (catong) [reply en_us/olx/source/components/problem-components.rst:8]: Yeah, that is weird. I always make sure I have the latest master before branching. The file is correct, it just seems that the paragraph before the contents directive is missing from the output even though it's in the source file.

**Comment by catong** [comment]
Merging https://github.com/edx/edx-documentation/pull/1229

**Pull request merged**